src/eventsource.ts

```diff
@@ -82,6 +91,9 @@ export default class CustomEventSource {
     this.closeEventSource()
     this.clearTimers()
     this.isClosed = true
+    if (typeof this.esOnClose === 'function') {
+      this.esOnClose()
+    }
```

Comment:
Can this be another thing than `undefined` or `null`? Otherwise this can be:
```suggestion
      this.esOnClose?.()
```